src/Core/SqlManager/Form/SqlRequestFormFactory.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\SqlManager\Form;
+
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObjectFormDataProviderInterface;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObjectFormFactoryInterface;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Form\Admin\Configure\AdvancedParameters\RequestSql\SqlRequestType;
+use Symfony\Component\Form\FormFactoryInterface;
+
+/**
+ * Class SqlRequestFormFactory creates form used in Back Office for adding/editing SqlRequests.
+ */
+final class SqlRequestFormFactory implements IdentifiableObjectFormFactoryInterface
```

Comment:
This class looks like `PrestaShop\PrestaShop\Core\Form\FormHandler`, but not generic. Do you think we can make a generic version for CRUD pages ?

Reply:
yes, i want to make it more generic as well, but i still didnt figure out what would be the best way to configure it (define form type, add form options, set form data). :/

Reply:
- define form type: same as Configuration Form handler https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Form/FormHandler.php#L103
- define form options: Configuration Form handler does not handle them, do we need some for CRUD ? in this example, no
- set form data: you take the data from given `IdentifiableObjectFormDataProviderInterface` and set it, just the way you do

What I suggest: we merge this, migrate another CRUD page and compare the FormFactories. If we can, we merge the 2 FormFactories into 1 generic class. Wdyt ?

Reply:
the form types worked fine for `FormHandler`, and its `Symfony` like, why not use them right now?